keras_core/models/functional.py

```diff
@@ -384,7 +386,10 @@ def get_config(self):
                     if node_data is not None:
                         filtered_inbound_nodes.append(node_data)
 
-            layer_config = serialization_lib.serialize_keras_object(operation)
+            serialize_obj_fn = serialization_lib.serialize_keras_object
+            if getattr(self, "use_legacy_config", False):
```

Comment:
When is this set to True? It seems somewhat unsafe to use a model-level attribute since the attribute will leak. I wonder if there could be a better way? A scope?

Reply:
Replaced `use_legacy_config` attr with a scope in new file saving_options.py, which avoids circular imports and will be expanded with SavedModel options utilities.